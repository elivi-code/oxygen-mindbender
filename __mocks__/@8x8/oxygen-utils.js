export * from '@8x8/oxygen-utils';

const useFontSize = jest.fn(() => 16);

export { useFontSize };
