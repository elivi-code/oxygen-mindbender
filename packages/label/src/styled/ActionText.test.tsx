import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import { label as theme } from '@8x8/oxygen-constants';

import ActionText from './ActionText';

describe('<ActionText>', () => {
  let wrapper;

  function setup(props = {}) {
    return renderer.create(<ActionText theme={theme} {...props} />).toJSON();
  }

  it('should render correctly', () => {
    wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when is disabled', () => {
    wrapper = setup({ isDisabled: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly the action text when expects to wrap', () => {
    wrapper = setup({ shouldWrapText: true });
    expect(wrapper).toMatchSnapshot();
  });
});
