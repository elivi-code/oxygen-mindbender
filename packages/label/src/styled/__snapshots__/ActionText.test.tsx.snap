// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionText> should render correctly 1`] = `
.c0 {
  font-size: inherit;
  text-align: left;
  background: transparent;
  border: none;
  outline: none;
  padding: 2px 4px;
  margin: 0 -4px 0 24px;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  border-radius: 2px;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  max-width: 100%;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  word-wrap: normal;
  cursor: pointer;
  color: #0376DA;
}

.c0:hover {
  color: #6FB1EA;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0:focus-visible {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  box-shadow: inset 0 0 0 2px #0376DA;
}

<a
  className="c0"
/>
`;

exports[`<ActionText> should render correctly the action text when expects to wrap 1`] = `
.c0 {
  font-size: inherit;
  text-align: left;
  background: transparent;
  border: none;
  outline: none;
  padding: 2px 4px;
  margin: 0 -4px 0 24px;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  border-radius: 2px;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  color: #0376DA;
}

.c0:hover {
  color: #6FB1EA;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0:focus-visible {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  box-shadow: inset 0 0 0 2px #0376DA;
}

<a
  className="c0"
/>
`;

exports[`<ActionText> should render correctly when is disabled 1`] = `
.c0 {
  font-size: inherit;
  text-align: left;
  background: transparent;
  border: none;
  outline: none;
  padding: 2px 4px;
  margin: 0 -4px 0 24px;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  border-radius: 2px;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  max-width: 100%;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  word-wrap: normal;
  color: #6FB1EA;
  cursor: not-allowed;
}

.c0:focus-visible {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  box-shadow: inset 0 0 0 2px #0376DA;
}

<a
  className="c0"
/>
`;
